### PR TITLE
Hotfix/config init validation

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Replace custom logic of endpoint validation with a package , Currently it is failing because of validation 

Issue:
```
❯ flytectl config init --host 345435433-80905779.ap-northeast-2.elb.amazonaws.com:81 --insecure
INFO[0000] [0] Couldn't find a config file []. Relying on env vars and pflags.
{"json":{},"level":"error","msg":"failed to initialize token source provider. Err: failed to fetch auth metadata. Error: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp: missing address\"","ts":"2021-12-28T23:37:58+09:00"}
{"json":{},"level":"warning","msg":"Starting an unauthenticated client because: can't create authenticated channel without a TokenSourceProvider","ts":"2021-12-28T23:37:58+09:00"}
{"json":{},"level":"info","msg":"Initialized Admin client","ts":"2021-12-28T23:37:58+09:00"}
Error: Please use a valid endpoint
{"json":{},"level":"error","msg":"Please use a valid endpoint","ts":"2021-12-28T23:37:58+09:00"}
```

Tests : https://github.com/flyteorg/flytectl/compare/hotfix/config-init-validation?expand=1#diff-58b2a13135e51eeb90a5af7e184f6a8a38bd65a57244131072807b000c9548a5R74

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [ ] Smoke tested
- [x] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by evalsocket** [comment]
@pmahindrakar-oss Need your review, It's ready

File: cmd/configuration/configuration.go [hunk 1]
```diff
@@ -94,8 +92,7 @@ func initFlytectlConfig(ctx context.Context, reader io.Reader) error {
 
 	if len(initConfig.DefaultConfig.Host) > 0 {
 		trimHost := trim(initConfig.DefaultConfig.Host)
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:94]: rename to trimmedEndpoint

File: cmd/configuration/configuration.go [hunk 2]
```diff
@@ -141,13 +138,27 @@ func trim(hostname string) string {
 }
 
 func validateEndpointName(domain string) bool {
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:140]: domain-> endPoint

File: cmd/configuration/configuration.go [hunk 3]
```diff
@@ -141,13 +138,27 @@ func trim(hostname string) string {
...
+	var validate = false
+	if domain == "localhost" {
 		return true
 	}
-	if net.ParseIP(domain) == nil {
+	if err := is.URL.Validate(domain); err != nil {
 		return false
 	}
-	return true
+	dns := strings.Split(domain, ":")
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:148]: endPointParts

File: cmd/configuration/configuration.go [hunk 4]
```diff
@@ -141,13 +138,27 @@ func trim(hostname string) string {
 }
 
 func validateEndpointName(domain string) bool {
-	RegExp := regexp.MustCompile(`^(([a-zA-Z]{1})|([a-zA-Z]{1}[a-zA-Z]{1})|([a-zA-Z]{1}[0-9]{1})|([0-9]{1}[a-zA-Z]{1})|([a-zA-Z0-9][a-zA-Z0-9-_]{1,61}[a-zA-Z0-9]))\.([a-zA-Z]{2,6}|[a-zA-Z0-9-]{2,30}\.[a-zA-Z
- ]{2,3})$`)
-	if RegExp.MatchString(domain) || domain == "localhost" {
+	var validate = false
+	if domain == "localhost" {
 		return true
 	}
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:144]: Instead of doing this explicitly , can any of the is package functions validate it. eg : is.DNSName.Validate

Comment (evalsocket) [reply cmd/configuration/configuration.go]: no it doesn't validate

**Review by pmahindrakar-oss — DISMISSED** [review]

**Review by pmahindrakar-oss — APPROVED** [review]
